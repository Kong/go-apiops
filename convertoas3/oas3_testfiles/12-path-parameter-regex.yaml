# The path parameters should be converted to a regex for matching the route.
# Including creating captures of the parameter values.

openapi: 3.0.3
info:
  title: Path parameter test
  version: v1
servers:
  - url: "https://example.com"

paths:
  # note that '-to-do' is not a valid regex capture name, hence it must be sanitized
  # in the output.
  /demo/{something}/else/{-to-do}/:
    get:
      operationId: opsid
      parameters:
        - in: path
          name: something
          required: true
          schema:
            type: string
        - in: path
          name: todo
          required: true
          schema:
            type: string
      responses:
        "200":
          description: OK
  # some more complex paths, with and without parameters
  # examples from: https://github.com/Kong/insomnia/pull/5246
  /batchs(Material='iron',Batch='10'):
    get:
      responses:
        "200":
          description: no parameters
      operationId: getBatchNoParams
  /batchs(Material='{Material}',Batch='{Batch}'):
    get:
      parameters:
        - in: path
          name: Material
          required: true
          schema:
            type: string
        - in: path
          name: Batch
          required: true
          schema:
            type: string
      responses:
        "200":
          description: An echo message.
      operationId: getBatchWithParams
