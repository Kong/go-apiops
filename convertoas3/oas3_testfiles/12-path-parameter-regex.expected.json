{
  "_format_version": "3.0",
  "services": [
    {
      "host": "example.com",
      "id": "520e4991-2d1a-59b8-bf2e-579cca0969a0",
      "name": "path-parameter-test",
      "path": "/",
      "plugins": [],
      "port": 443,
      "protocol": "https",
      "routes": [
        {
          "id": "38dcec89-d67c-5f67-ab5e-b38cc2180f77",
          "methods": [
            "GET"
          ],
          "name": "path-parameter-test_getbatchnoparams",
          "paths": [
            "~/batchs\\(Material='iron',Batch='10'\\)$"
          ],
          "plugins": [],
          "regex_priority": 200,
          "strip_path": false,
          "tags": [
            "OAS3_import",
            "OAS3file_12-path-parameter-regex.yaml"
          ]
        },
        {
          "id": "2af1536a-706a-5941-b3fe-2682e031edf9",
          "methods": [
            "GET"
          ],
          "name": "path-parameter-test_getbatchwithparams",
          "paths": [
            "~/batchs\\(Material='(?<material>[^#?/]+)',Batch='(?<batch>[^#?/]+)'\\)$"
          ],
          "plugins": [],
          "regex_priority": 100,
          "strip_path": false,
          "tags": [
            "OAS3_import",
            "OAS3file_12-path-parameter-regex.yaml"
          ]
        },
        {
          "id": "0da1f8dc-e918-5379-b3b0-ffc061ae1691",
          "methods": [
            "GET"
          ],
          "name": "path-parameter-test_opsid",
          "paths": [
            "~/demo/(?<something>[^#?/]+)/else/(?<to_do>[^#?/]+)/$"
          ],
          "plugins": [],
          "regex_priority": 100,
          "strip_path": false,
          "tags": [
            "OAS3_import",
            "OAS3file_12-path-parameter-regex.yaml"
          ]
        }
      ],
      "tags": [
        "OAS3_import",
        "OAS3file_12-path-parameter-regex.yaml"
      ]
    }
  ],
  "upstreams": []
}
